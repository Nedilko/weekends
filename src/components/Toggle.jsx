import PropTypes from 'prop-types'
import { useState } from 'react'

Toggle.propTypes = {
  label: PropTypes.string,
  checked: PropTypes.bool,
}

function Toggle({ label, checked = false }) {
  const [isChecked, setIsChecked] = useState(checked)
  const handleChange = (e) => {
    setIsChecked(e.target.checked)
  }
  return (
    <label className="relative flex items-center group p-2 text-xl font-thin text-slate-600 dark:text-gray-400">
      {label}
      <input
        type="checkbox"
        checked={isChecked}
        onClick={handleChange}
        className="absolute left-1/2 -translate-x-1/2 w-full h-full peer appearance-none rounded-md"
      />
      <span className="w-11 h-6 flex items-center flex-shrink-0 ml-3 p-1 bg-gray-300 rounded-full duration-300 ease-in-out peer-checked:bg-slate-500 after:w-5 after:h-5 after:bg-white after:rounded-full after:shadow-md after:duration-300 peer-checked:after:translate-x-4 group-hover:after:translate-x-0.5"></span>
    </label>
  )
}

export default Toggle
