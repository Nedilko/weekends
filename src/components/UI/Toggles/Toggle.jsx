import PropTypes from 'prop-types'

Toggle.propTypes = {
  label: PropTypes.string,
  isChecked: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
}

function Toggle({ label, isChecked, onClick }) {
  return (
    <label className="flex items-center px-1 text-xl font-thin text-slate-600 dark:text-gray-400">
      {label}
      <span
        role="checkbox"
        onClick={onClick}
        className={`ml-4 flex h-6 w-12 flex-shrink-0 items-center justify-between rounded-full px-1 duration-300 ease-in-out ${
          isChecked
            ? 'bg-green-400 dark:bg-green-600'
            : 'bg-gray-200 dark:bg-slate-700'
        }`}
      >
        <span
          className={`absolute flex h-5 w-5 rounded-full bg-white text-yellow-500 shadow-md duration-300 ease-in-out dark:bg-slate-900 ${
            isChecked ? 'translate-x-5' : ''
          }`}
        ></span>
      </span>
    </label>
  )
}

export default Toggle
