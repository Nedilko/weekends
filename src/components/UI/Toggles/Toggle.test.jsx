import Toggle from '@UI/Toggles/Toggle'
import { render, screen, userEvent } from '@utils/test-utils'
import { describe, expect, it, vi } from 'vitest'

describe('toggle is', () => {
  const clickHandler = vi.fn()
  beforeEach(() => {
    render(<Toggle isChecked={true} onClick={clickHandler} />)
  })
  it('in the document', () => {
    const toggle = screen.getByRole('checkbox')
    expect(toggle).toBeInTheDocument()
    expect(toggle).toMatchSnapshot()
  })
  it('clicked once', async () => {
    const user = userEvent.setup()
    const toggle = screen.getByRole('checkbox')
    await user.click(toggle)
    expect(clickHandler).toHaveBeenCalledTimes(1)
    expect(toggle).toMatchSnapshot()
  })
})
