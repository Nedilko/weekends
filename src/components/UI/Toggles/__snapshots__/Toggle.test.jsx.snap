// Vitest Snapshot v1

exports[`toggle is > in the document 1`] = `
<span
  class="ml-4 flex h-6 w-12 flex-shrink-0 items-center justify-between rounded-full px-1 duration-300 ease-in-out bg-gray-200 dark:bg-slate-700"
  role="checkbox"
>
  <span
    class="absolute flex h-5 w-5 rounded-full bg-white text-yellow-500 shadow-md duration-300 ease-in-out dark:bg-slate-900 "
  />
</span>
`;
